src/volume_mgr.rs

```diff
@@ -319,13 +353,55 @@ where
                 return Err(Error::VolumeStillInUse);
             }
         }
-
         let volume_idx = self.get_volume_by_id(volume)?;
+        if !self.open_volumes[volume_idx].read_only {
+            let VolumeType::Fat(volume_type) = &self.open_volumes[volume_idx].volume_type;
+            self.set_volume_status_dirty(volume_type, false)?;
+        }
+
         self.open_volumes.swap_remove(volume_idx);
 
         Ok(())
     }
 
+    /// Sets the volume status dirty to dirty if true, to not dirty if false
+    fn set_volume_status_dirty(
+        &self,
+        volume: &FatVolume,
+        dirty: bool,
+    ) -> Result<(), Error<D::Error>> {
+        let mut blocks = [Block::new()];
+        let fat_table1_start = volume.lba_start + volume.fat_start;
+        self.block_device
+            .read(&mut blocks, fat_table1_start, "reading fat table")?;
+        let block = &mut blocks[0];
+        let mut fat_table =
+            fat::FatTable::create_from_bytes(&mut block.contents, volume.get_fat_type())
+                .map_err(Error::FormatError)?;
+        fat_table.set_dirty(dirty);
+        if volume.fat_nums == 1 || volume.fat_nums == 2 {
+            self.block_device.write(&blocks, fat_table1_start)?;
+            // Synchronize also backup fat table
+            if volume.fat_nums == 2 {
+                self.block_device
+                    .write(&blocks, fat_table1_start + volume.fat_size)?
+            }
+        }
+        Ok(())
+    }
+    #[cfg(test)]
```

Comment:
Tests go in `mod test {}`.

Reply:
changed in 9d6865a75355be25cd1b3bbc5597d29d2862591c